velox/exec/tests/RowContainerTest.cpp

```diff
@@ -278,7 +279,44 @@ class RowContainerTest : public exec::test::RowContainerTestBase {
   }
 
   template <typename T>
-  void testCompareFloats(TypePtr type, bool ascending, bool nullsFirst) {
+  void setupFloatTestExpectedOutput(
+      T& initialOrder,
+      bool ascending,
+      bool nullsFirst) {
+    if (!ascending) {
+      if (nullsFirst) {
+        std::reverse(initialOrder.begin() + 1, initialOrder.end());
+      } else {
+        std::reverse(initialOrder.begin(), initialOrder.end());
+      }
+    } else {
+      if (!nullsFirst) {
+        for (int i = 0; i < initialOrder.size() - 1; ++i) {
+          initialOrder[i] = initialOrder[i + 1];
+        }
+        initialOrder[initialOrder.size() - 1] = std::nullopt;
+      }
+    }
+  }
+
+  template <typename T>
+  VectorPtr createResultVectorForSimpleComparison(
+      const std::vector<std::pair<int, char*>>& indexedRows,
+      const DecodedVector& decoded,
+      facebook::velox::test::VectorMaker& vectorMaker) {
+    std::vector<std::optional<T>> sorted;
+    for (const auto& irow : indexedRows) {
+      if (decoded.isNullAt(irow.first)) {
+        sorted.push_back(std::nullopt);
+      } else {
+        sorted.push_back(decoded.valueAt<T>(irow.first));
+      }
+    }
+    return vectorMaker.template flatVectorNullable<T>(sorted);
```

Comment:
An instance of VectorMaker is available in the base class (vectorMaker_). Also, there is a method makeNullableFlatVector<T>(sorted) available in the base class as well.

Reply:
Right, left over from the previous test. I didn't touch much of that code other than trying to refactor obvious common code.

Reply:
It would be nice to clean that up. Any chance you can send a refactoring PR?